common/aws/dynamodb/client.go

```diff
@@ -166,6 +166,22 @@ func (c *Client) QueryIndex(ctx context.Context, tableName string, indexName str
 	return response.Items, nil
 }
 
+// QueryCount returns the count of the items in the index that match the given key
+func (c *Client) QueryIndexCount(ctx context.Context, tableName string, indexName string, keyCondition string, expAttributeValues ExpresseionValues) (int32, error) {
```

Comment:
can we add a test for this method in https://github.com/Layr-Labs/eigenda/blob/master/common/aws/dynamodb/client_test.go?